Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp

```diff
@@ -1099,6 +1118,37 @@ WebMParser::ConsumeFrameDataResult WebMParser::VideoTrackData::consumeFrameData(
 #endif
 }
 
+void WebMParser::VideoTrackData::consumeAdditionalBlockData(const webm::BlockAdditions& additions)
+{
+    for (auto& blockMore : additions.block_mores) {
+        if (!blockMore.is_present())
+            continue;
+
+        // https://www.webmproject.org/docs/container/#BlockAddID
+        // 0x04 indicates ITU T.35 metadata as defined by ITU-T T.35 terminal codes.
+        if (!blockMore.value().data.is_present() || !blockMore.value().id.is_present() || blockMore.value().id.value() != 0x4)
+            continue;
+
+        // ANSI/CTA-861-H:
+        // S.3 User_data_registered_itu_t_t35 SEI message semantics for ST 2094-40 [55]
+        static constexpr std::array<uint8_t, 5> s_ST2094_40_Message = { 0xB5, 0x00, 0x3C, 0x00, 0x01 };
+
+        auto& blockMoreData = blockMore.value().data.value();
+        if (blockMoreData.size() < s_ST2094_40_Message.size())
+            continue;
+
+        auto blockMoreDataStart = std::span { blockMoreData }.subspan(0, s_ST2094_40_Message.size());
+        if (!std::ranges::equal(s_ST2094_40_Message, blockMoreDataStart))
+            continue;
+
+        auto hdrMetadata = FragmentedSharedBuffer::create(std::span { blockMore.value().data.value() });
+        for (auto& sample : m_pendingMediaSamples) {
+            sample.hdrMetadata = hdrMetadata.ptr();
```

Comment:
If we replaced the `Ref` above (or `auto` as written) we don't have to pass the ptr().

Reply:
This should just be a `WTFMove()` upon further reflection.